keps/sig-instrumentation/1661-event-api-migration/README.md

```diff
@@ -0,0 +1,478 @@
+
+  // ID of the controller instance, e.g. `kubelet-xyzf`.
+  // +optional
+  ReportingInstance string `json:"reportingInstance,omitempty" protobuf:"bytes,5,opt,name=reportingInstance"`
+
+  // What action was taken/failed regarding to the regarding object.
+  // +optional
+  Action string `json:"action,omitempty" protobuf:"bytes,6,name=action"`
+
+  // Why the action was taken.
+  Reason string `json:"reason,omitempty" protobuf:"bytes,7,name=reason"`
+
+  // The object this Event is about. In most cases it's an Object reporting controller implements.
+  // E.g. ReplicaSetController implements ReplicaSets and this event is emitted because
+  // it acts on some changes in a ReplicaSet object.
+  // +optional
+  Regarding corev1.ObjectReference `json:"regarding,omitempty" protobuf:"bytes,8,opt,name=regarding"`
+
+  // Optional secondary object for more complex actions. E.g. when regarding object triggers
+  // a creation or deletion of related object.
+  // +optional
+  Related *corev1.ObjectReference `json:"related,omitempty" protobuf:"bytes,9,opt,name=related"`
+
+  // Optional. A human-readable description of the status of this operation.
+  // Maximal length of the note is 1kB, but libraries should be prepared to
+  // handle values up to 64kB.
+  // +optional
+  Note string `json:"note,omitempty" protobuf:"bytes,10,opt,name=note"`
+
+  // Type of this event (Normal, Warning), new types could be added in the
+  // future.
+  // +optional
+  Type string `json:"type,omitempty" protobuf:"bytes,11,opt,name=type"`
+
+  // Deprecated field assuring backward compatibility with core.v1 Event type
+  // +optional
+  DeprecatedSource corev1.EventSource `json:"deprecatedSource,omitempty" protobuf:"bytes,12,opt,name=deprecatedSource"`
+  // Deprecated field assuring backward compatibility with core.v1 Event type
+  // +optional
+  DeprecatedFirstTimestamp metav1.Time `json:"deprecatedFirstTimestamp,omitempty" protobuf:"bytes,13,opt,name=deprecatedFirstTimestamp"`
+  // Deprecated field assuring backward compatibility with core.v1 Event type
+  // +optional
+  DeprecatedLastTimestamp metav1.Time `json:"deprecatedLastTimestamp,omitempty" protobuf:"bytes,14,opt,name=deprecatedLastTimestamp"`
+  // Deprecated field assuring backward compatibility with core.v1 Event type
+  // +optional
+  DeprecatedCount int32 `json:"deprecatedCount,omitempty" protobuf:"varint,15,opt,name=deprecatedCount"`
+}
+
+// EventSeries contain information on series of events, i.e. thing that was/is happening
+// continuously for some time.
+type EventSeries struct {
+  // Number of occurrences in this series up to the last heartbeat time
+  Count int32 `json:"count" protobuf:"varint,1,opt,name=count"`
+  // Time when last Event from the series was seen before last heartbeat.
+  LastObservedTime metav1.MicroTime `json:"lastObservedTime" protobuf:"bytes,2,opt,name=lastObservedTime"`
+  // Information whether this series is ongoing or finished.
+  // Deprecated. Planned removal for 1.18
```

Comment:
nit: I think the deprecation notice here is a mistake?

Reply:
It's not. We initially added it, but then realized it was a mistake and deprecated it in 1.14 (yes year ago+).

Reply:
Oh, duh.  I just realized this KEP is just for the migration, not proposing API changes.  It all makes sense now.